src/views/NoticeOfWithdrawal.vue

```diff
@@ -0,0 +1,1045 @@
+    @Getter(useRootStore) getBusinessEmail!: string
+    @Getter(useBusinessStore) getLegalName!: string
+    @Getter(useConfigurationStore) getPayApiUrl!: string
+    @Getter(useRootStore) getUserInfo!: any
+    @Getter(useRootStore) isRoleStaff!: boolean
+
+    // enum for template
+    readonly FilingCodes = FilingCodes
+
+    // variables for Certify component
+    certifiedBy = ''
+    isCertified = false
+    certifyFormValid = false
+
+    // variables for POA component
+    hasPlanOfArrangement = false
+    hasComeIntoEffect = false
+
+    // variable for Reference Number component
+    referenceNumberValid = true
+    referenceNumber = ''
+
+    // variables for Document Delivery component
+    documentDeliveryValid = true
+    documentOptionalEmail = ''
+
+    // variables for staff payment
+    staffPaymentValid = false
+    staffPaymentData = { option: StaffPaymentOptions.NONE } as StaffPaymentIF
+
+    // variables for displaying dialogs
+    staffRoleErrorDialog = false
+    resumeErrorDialog = false
+    saveErrorReason = null as SaveErrorReasons
+    paymentErrorDialog = false
+
+    // other variables
+    totalFee = 0
+    dataLoaded = false
+    loadingMessage = ''
+    filingId = 0 // id of this Notice if Withdrawal filing
+    savedFiling: any = null // filing during save
+    saving = false // true only when saving
+    savingResuming = false // true only when saving and resuming
+    showErrors = false // true when we press on File and Pay (trigger validation)
+    filingPaying = false // true only when filing and paying
+    haveChanges = false
+    saveErrors = []
+    saveWarnings = []
+    filingToBeWithdrawn = ''
+
+    /** True if loading container should be shown, else False. */
+    get showLoadingContainer (): boolean {
+      // show loading container when data isn't yet loaded and when
+      // no dialogs are displayed (otherwise dialogs may be hidden)
+      return (!this.dataLoaded && !this.saveErrorReason && !this.paymentErrorDialog)
+    }
+
+    /** The Base URL string. */
+    get baseUrl (): string {
+      return sessionStorage.getItem('BASE_URL')
+    }
+
+    /** True if page is valid, else False. */
+    get isPageValid (): boolean {
+      return (
+        this.certifyFormValid &&
+        this.documentDeliveryValid &&
+        this.referenceNumberValid &&
+        this.staffPaymentValid
+      )
+    }
+
+    /** True when saving, saving and resuming, or filing and paying. */
+    get busySaving (): boolean {
+      return (this.saving || this.savingResuming || this.filingPaying)
+    }
+
+    /** Called when component is created. */
+    created (): void {
+      // do not proceed if user is not staff
+      if (!this.isRoleStaff) {
+        this.staffRoleErrorDialog = true
+        throw new Error('This is a Staff only Filing.')
+      }
+
+      // init
+      this.setFilingData([])
+      this.filingToBeWithdrawn = this.$route.query.filingToBeWithdrawn as string || ''
+
```

Comment:
like: BC0883739/notice-of-withdrawal?filingToBeWithdrawn=153004&filingId=0

Reply:
OK.

You could move it near the other route query parameter (line 455).

Are you sure id should be a string?

Reply:
It should be a string. But I changed the type to any, just to be safe